lib/src/http/auto_retry.dart

```diff
@@ -0,0 +1,105 @@
+library w_transport.src.http.auto_retry;
+
+import 'package:w_transport/src/http/base_request.dart';
+import 'package:w_transport/src/http/finalized_request.dart';
+import 'package:w_transport/src/http/request_exception.dart';
+import 'package:w_transport/src/http/requests.dart';
+import 'package:w_transport/src/http/response.dart';
+
+typedef /* Future<bool> | bool */ RetryTest(
+    FinalizedRequest request, BaseResponse response);
```

Comment:
Yeah, I agree :/ Almost all use cases I can think of could be synchronous (async would only be necessary if a streamed response needed to be read), but it is simpler to just require async.